src/exchanges.js

```diff
@@ -0,0 +1,75 @@
+let base = require('taskcluster-base');
+
+let exchanges = new base.Exchanges({
+  title: 'S3 Distribute Pulse Exchanges',
+  description: [
+    'Exchanges from the s3 distribution manager',
+  ].join('\n'),
+  schemaPrefix: 'http://schemas.taskcluster.net/aws-provisioner/v1/',
+});
+
+module.exports = exchanges;
+
+let commonRoutingKey = [
+  {
+    // Let's keep the "primary." prefix, so we can support custom routing keys
+    // in the future, I don't see a need for it here. But it's nice to have the
+    // option of adding it later...
+    name: 'routingKeyKind',
+    summary: 'Identifier for the routing-key kind. This is ' +
+             'always `\'primary\'` for the formalized routing key.',
+    constant: 'primary',
+    required: true,
+  }, {
+    name: 'service',
```

Comment:
Hmm.  I think there's nice properties to not having to do string manipulation in order to figure out a service.  We can still make things generic and have separate services and regions.  Just like we can do gcs:<region>, we could do gcs.<region> or in api /:service/:region